src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -418,13 +418,22 @@ private void respond(ChannelHandlerContext ctx, int reqSeq, FullHttpRequest req,
 
         assert status.code() >= 400;
 
-        final String msg = errorMessage(status);
-        if (cause != null) {
-            logger.warn("{} Unexpected failure: {}", ctx.channel(), msg, cause);
+        final ByteBuf content;
+        if (req.method() == HttpMethod.HEAD) {
+            // A response to a HEAD request must have no content.
+            content = Unpooled.EMPTY_BUFFER;
+            if (cause != null) {
+                Exceptions.logIfUnexpected(logger, ctx.channel(), errorMessage(status), cause);
+            }
+        } else {
+            final String msg = errorMessage(status);
```

Comment:
We still have to call `errorMessage(status)` only when `cause != null`. The resulting code would be:

```java
    String errorMsg;
    if (req.method() == HttpMethod.HEAD) {
        ...
        if (cause != null) {
            errorMsg = errorMessage(status);
        } else {
            errorMsg = null;
        }
    } else {
        ...
        errorMsg = msg;
    }

    if (cause != null) {
        Exceptions.logIfUnexpected(logger, ctx.channel(), errorMsg, cause);
    }
```
I'm not sure if this is simpler because it adds another local variable and `if`. Do you have a better idea to make this simpler?